libraries/HID/src/HID.cpp

```diff
@@ -64,12 +82,23 @@ int HID_::getDescriptor(USBSetup& setup)
 
 uint8_t HID_::getShortName(char *name)
 {
+    if(serial) {
+        for(byte i=0; i<strlen_P(serial); i++) {
+            name[i] = pgm_read_byte_near(serial + i);
+        }
+        return strlen_P(serial);
+    }
```

Comment:
#83 should also be considered here. I currenty use a workaround in one of my projects https://github.com/Legion2/CorsairLightingProtocol/commit/2320e5bdb22141a2678612842ce229c3bbe42731

Reply:
Agree with you, the cycle needs to be limited by ISERIAL_MAX_LEN, to avoid memory overflow. Will fix and resubmit.

Reply:
>Will fix and resubmit.

@abratchik any commit you make to [the `master` branch of your fork](https://github.com/abratchik/ArduinoCore-avr) will automatically be added to this pull request. So you can continue to make improvements to this pull request rather than closing this and making a new one. that will allow all relevant discussion to be consolidated in a single place.

Reply:
@per1234 thank you, just added another commit for fixing the issue #83.

Reply:
> #83 should also be considered here. I currenty use a workaround in one of my projects [Legion2/CorsairLightingProtocol@2320e5b](https://github.com/Legion2/CorsairLightingProtocol/commit/2320e5bdb22141a2678612842ce229c3bbe42731)

Fixed in d9b18f4